src/commands.ts

```diff
@@ -63,11 +63,22 @@ function applySuggestionsToTransform(
     if (isToRevert) {
       const { pos: deletionFrom, deleted } = tr.mapping.mapResult(pos);
       if (deleted) return false;
+
       const deletionTo = findSuggestionMarkEnd(
         tr.doc.resolve(deletionFrom + child.nodeSize),
         markTypeToRevert,
       );
-      tr.deleteRange(deletionFrom, deletionTo);
+      // check if the previous and the next text part is a space
+      // if so, we can delete the whole text part
+      const prevChar = tr.doc.textBetween(
+        deletionFrom - 1,
+        deletionFrom,
+        "x",
+        "x",
+      );
+      const nextChar = tr.doc.textBetween(deletionTo, deletionTo + 1, "x", "x");
```

Comment:
I think these will blow up if at the very beginning or very end of the document, no? We may need to first confirm that there _is_ a `deletionFrom - 1` and `deletionTo + 1`.